Azure/azure-cli
# [KeyVault] Support private endpoint connections

Support for #11038 
Bump the Key Vault mgmt-plane SDK version to 2.1.1 (This is the fixed version, 2.1.0 is buggy)

Add two command groups, all of them are marked as `preview`:
1. `az keyvault private-endpoint`: manage vault private endpoint connections.
2. `az keyvault private-link-resource`: manage vault private link resources.

Whole scenario and test instructions:
1. Prepare the resources: a `Key Vault` and a `Virtual Network` within the same region (preferred region: `Central US EUAP`).
2. Disable private endpoint network policy for the new created vnet: https://docs.microsoft.com/en-us/azure/private-link/disable-private-endpoint-network-policy#using-azure-cli
3. Create a private endpoint connection request via network command: `network private-endpoint create -g {rg} -n {pe} --vnet-name {vnet} --subnet {subnet} -l {loc} --connection-name {pe_connection} --private-connection-resource-id {kv_id} --group-ids vault` (The `group-ids` can be obtained from command `az keyvault private-link-resource show`, for Key Vault service, the group id is `vault`, I remember that @MyronFanQiu planned to implement a new command in `network` group, which can be used to obtain the group id of all service types).
4. Show the private endpoint connection on Key Vault side: `keyvault private-endpoint show --connection-id {kv_pe_id}`.
5. Approve or reject this connection request: `keyvault private-endpoint approve -n {kv} --connection-name {kv_pe_name} --approval-description "{approval_desc}"`, `keyvault private-endpoint reject --connection-id {kv_pe_id} --rejection-description "{rejection_desc}"`

---
This checklist is used to make sure that common guidelines for a pull request are followed.
